sentry-android-core/src/test/java/io/sentry/android/core/DefaultAndroidEventProcessorTest.kt

```diff
@@ -290,4 +290,26 @@ class DefaultAndroidEventProcessorTest {
         assertSame(osNoName, (event.contexts["os_1"] as OperatingSystem))
         assertEquals("Android", event.contexts.operatingSystem!!.name)
     }
+
+    @Test
+    fun `When hint is Cached, memory data should not be applied`() {
+        val sut = fixture.getSut(context)
+
+        var event = SentryEvent()
+        event = sut.process(event, CachedEvent())
+
+        assertNull(event.contexts.device!!.freeMemory)
+        assertNull(event.contexts.device!!.isLowMemory)
+    }
+
+    @Test
+    fun `When hint is not Cached, memory data should be applied`() {
+        val sut = fixture.getSut(context)
+
+        var event = SentryEvent()
+        event = sut.process(event, null)
+
+        assertNotNull(event.contexts.device!!.freeMemory)
+        assertNotNull(event.contexts.device!!.isLowMemory)
```

Comment:
`l`: Do we know the expected values for `freeMemory` and `isLowMemory`? If so, we could assert against them. This would make the test a bit better. Otherwise, just ignore me.

Reply:
no, it can vary depending on robolectric tests